dl1_data_handler/writer.py

```diff
@@ -465,52 +645,139 @@ def _create_event_table(self, subarray):
-                table.flush()
+                image_table.flush()
 
-    def prepare_file(self, input_filename, subarray, mcheader):
+    def _create_parameter_tables(self, subarray, index_parameters_table, cleaning_main_algorithm_metadata):
+        self.file.create_group(self.file.root, "Parameters"+str(index_parameters_table))
+        for tel_desc in set(subarray.tels.values()):
+            tel_name = str(tel_desc)
+            parameter_table_number = getattr(self.file.root, "Parameters" + str(index_parameters_table))
+            if ("/{}".format(tel_name)) not in parameter_table_number:
+                logger.info("Creating {} parameter table...".format(tel_name))
+
+                # Calculate expected number of rows for compression
+                if tel_name in self.expected_images_per_event:
+                    expected_rows = (
+                        self.expected_events * self.expected_images_per_event[
+                            tel_name])
+                else:
+                    expected_rows = (
+                        self.DEFAULT_IMGS_PER_EVENT * self.expected_events)
+
+                if index_parameters_table == 0:
+                    parameter_table = self.file.create_table(parameter_table_number,
+                                                         tel_name,
+                                                         table_defs.ParametersTableRow,
+                                                         "Parameter table of {} parameters, algorithm: {}, args: {}".format(tel_name,
+                                                                                           cleaning_main_algorithm_metadata['algorithm'],
+                                                                                           cleaning_main_algorithm_metadata['args']),
+                                                         filters=self.filters,
+                                                         expectedrows=expected_rows)
+                else :
+                    parameter_table = self.file.create_table(parameter_table_number,
+                                                         tel_name,
+                                                         table_defs.ParametersTableRow,
+                                                         "Parameter table of {} parameters, algorithm: {}, args: {}".format(tel_name,
+                                                                  self.cleaning_settings[index_parameters_table -1]['algorithm'],
+                                                                  self.cleaning_settings[index_parameters_table -1]['args']),
+                                                        filters=self.filters,
+                                                        expectedrows=expected_rows)
+
+                # Place blank image at index 0 of all image tables
+                parameter_row = parameter_table.row
+
+                parameter_row['event_index'] = -1
```

Comment:
better to initialize values to `np.nan` or `None`, see the initialization of all these quantities in:
https://github.com/cta-observatory/cta-lstchain/blob/master/lstchain/io/lstcontainers.py

Reply:
I agree, it's better to use `np.nan` for initialization, but those are stored in the file, i.e. the data types have to match the with the real quantities. It's needed to construct the tables before.

Reply:
Solved. Thanks for the hint @rlopezcoto :)

Reply:
That is an integer, -1 is the correct choice as there is no nan for integers and putting nan will result in the value being stored in the wrong data type.